File: forest/stillness-in-motion.clj
(defn goback []
  (.wait this 0.001)
  (if (and (= (.x (.pos this)) 40) (= (.y (.pos this)) 34))
    (.shield this)
    (.moveXY this 40 34)))

(while true
  (def e (.findNearestEnemy this))
  (if e
    (if (< (.distanceTo this e) 5)
      (.attack this e)
      (goback))
    (goback)))

File: forest/stillness-in-motion.lua
loop
    local e = self:findNearest(self:findEnemies())
    if e then
        if self:distanceTo(e) < 5 and self:isReady("bash") then
            self:bash(e)
        elseif self.pos.x == 40 and self.pos.y == 34 then
            self:shield()
        else
            self:move({x=40, y=34})
        end
    end
    if self.gold >= self:costOf("arrow-tower") then
        self:buildXY("arrow-tower", 34, 34)
    end
end
